In [2]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())

options = Options()
options.add_argument("headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://davelee-fun.github.io")

element = driver.find_element(By.TAG_NAME, "h1")
print(element.text)

time.sleep(3)

driver.quit()

Teddy의 선물 블로그


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app
- 부모요소의 id 속성 값이 dev_course_list인 자식요소의 class가 course인 데이터만 출력해주세요.

In [12]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())

options = Options()
options.add_argument("headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://crawlingtest.netlify.app/")

elements = driver.find_elements(By.CSS_SELECTOR, "ul#dev_course_list > li.course")

for element in elements :
    print(element.text)

time.sleep(3)

driver.quit()

(초급) - 강사가 실제 사용하는 자동 프로그램 소개 [2]
(초급) - 필요한 프로그램 설치 시연 [5]
(초급) - 데이터를 엑셀 파일로 만들기 [9]
(초급) -     엑셀 파일 이쁘게! 이쁘게! [8]
(초급) -     나대신 주기적으로 파이썬 프로그램 실행하기 [7]
(초급) - 파이썬으로 슬랙(slack) 메신저에 글쓰기 [40]
(초급) - 웹사이트 변경사항 주기적으로 체크해서, 메신저로 알람주기 [12]
(초급) - 네이버 API 사용해서, 블로그에 글쓰기 [42]
(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app
- 해당 사이트 내 커리큘럼 일정 표 안에 있는 데이터를 모두 추출해서 출력해주세요.

In [35]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())

options = Options()
options.add_argument("headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://crawlingtest.netlify.app/")

# elements = driver.find_elements(By.TAG_NAME, "td")
elements = driver.find_elements(By.CSS_SELECTOR, "table tr td")

for element in elements :
    print(element.text)
    
time.sleep(3)

driver.quit()

일정
커리큘럼 타이틀
난이도
5.1 ~ 6.15
나만의 엣지있는 블로그 사이트 만들기 (취미로 익히는 IT)
초급
6.16 ~ 7.31
파이썬과 데이터과학 첫걸음 (IT 기본기 익히기)
중급


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app
- 아래 데이터만 추출해오세요.
- (중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]

In [57]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())

options = Options()
options.add_argument("headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://crawlingtest.netlify.app/")

elements = driver.find_elements(By.CSS_SELECTOR, "ul#dev_course_list > li.course")

# for element in elements :
#     print(element.text)

print(elements[8].text)
    
time.sleep(3)

driver.quit()

(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- class 값이 "news"인 요소의 태그 내용을 찾아서 출력

In [110]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())

options = Options()
options.add_argument("headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://divjason.github.io/sellenium-test/")

element = driver.find_element(By.CSS_SELECTOR, "span.news")
print(element.text)
    
time.sleep(3)

driver.quit()

뉴스: 머신러닝 강좌가 나왔어요~~~


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로 클릭하기
- 클릭 이후 이동한 페이지에서 message 라는 클래스 속성값을 가지고 있는 요소의 텍스트 정보를 출력

In [122]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())

options = Options()
# options.add_argument("headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36") #--User-agent= 윈도우 버전. 추가하면 헤드리스랑 겹쳐서 사용가능
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://divjason.github.io/sellenium-test/")

# 내가 한 거
# element = driver.find_element(By.NAME, "username")
# element.clear()
# element.send_keys("error@error.com")

# element = driver.find_element(By.NAME, "password")
# element.clear() # 초기화 먼저
# element.send_keys("1234")

# element.send_keys(Keys.ENTER) # RETURN과 ENTER가 똑같음

# 쌤 꺼
element = driver.find_element(By.ID, "username")
element.clear()
element.send_keys("error@error.com")

element = driver.find_element(By.ID, "password")
element.clear()
element.send_keys("1234")

time.sleep(5)

# element.send_keys(Keys.RETURN) # 이거 안먹는 사이트 무조건 있음. -> 명확한 버튼 값을 찾아와서 -- . 이건 야매
element = driver.find_element(By.CSS_SELECTOR, "input[type='submit']") # 속성 선택자
element.click() # 얘가 정석. 

time.sleep(5)

driver.quit()

#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로
  클릭 이후 이동한 페이지에서 message 라는 클래스 속성값을 가지고 있는 요소의 텍스트 정보를 출력

In [125]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())

options = Options()
# options.add_argument("headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36") #--User-agent= 윈도우 버전. 추가하면 헤드리스랑 겹쳐서 사용가능
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://divjason.github.io/sellenium-test/")

# 내가 한 거
# element = driver.find_element(By.NAME, "username")
# element.clear()
# element.send_keys("error@error.com")

# element = driver.find_element(By.NAME, "password")
# element.clear() # 초기화 먼저
# element.send_keys("1234")

# element.send_keys(Keys.ENTER) # RETURN과 ENTER가 똑같음

# 쌤 꺼
element = driver.find_element(By.ID, "username")
element.clear()
element.send_keys("error@error.com")

element = driver.find_element(By.ID, "password")
element.clear()
element.send_keys("1234")

time.sleep(3)

element = driver.find_element(By.CSS_SELECTOR, "input[type='submit']")
element.click()

time.sleep(3)

element = driver.find_element(By.CSS_SELECTOR, "div.message")
print(element.text)

driver.quit()

실제 사이트는 https://megaitlab.com/ 입니다
본 페이지는 로그인된 페이지입니다!
뉴스: 머신러닝 강좌가 나왔어요~~~


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로
  클릭 이후 이동한 페이지에서 message 라는 클래스 속성값을 가지고 있는 요소의 텍스트 정보를 출력
- 해당 페이지에서 로그아웃 버튼을 클릭 후 이동한 페이지에서 다시 message 라는 클래스 속성값을 가지고 있는 요소의 텍스트 정보를 출력

In [1]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
import time
import pyautogui

service = Service(ChromeDriverManager().install())

options = Options()
# options.add_argument("headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36") #--User-agent= 윈도우 버전. 추가하면 헤드리스랑 겹쳐서 사용가능
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://divjason.github.io/sellenium-test/")

element = driver.find_element(By.ID, "username")
element.clear()
element.send_keys("error@error.com")

element = driver.find_element(By.ID, "password")
element.clear()
element.send_keys("1234")

time.sleep(3)

element = driver.find_element(By.CSS_SELECTOR, "input[type='submit']")
element.click()

time.sleep(3)

element = driver.find_element(By.CSS_SELECTOR, "div.message")

element = driver.find_element(By.CSS_SELECTOR, "input[type='submit']")
element.click()

time.sleep(3)

element = driver.find_element(By.CSS_SELECTOR, "div.message")
print(element.text)

driver.quit()

WebDriverException: Message: Service /Users/hyelincho/.wdm/drivers/chromedriver/mac64/138.0.7204.168/chromedriver-mac-arm64/chromedriver unexpectedly exited. Status code was: -9
